### PR TITLE
Complete Rewrite, no new features

### PR DESCRIPTION
**Pull request by hybras**
* Use 2018 rust (no need for `extern $crate`)
* Clap v3
* notify-rust v4
* Clap derive (increases binary size)
  * Refactor linux specific args into separate struct
* Use conditional compilation instead of "not implemented" / "unsupported" error messages

Tested on M1 mac, macOS 12.1

This is a continuation of #10

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,16 +1,17 @@
 [package]
+edition = '2018'
 name = "toastify"
 version = "0.4.0"
 authors = ["Hendrik Sollich <hendrik@hoodie.de>"]
-description = "A commandline tool that shows desktop notifications using [notify-rust](http://hoodie.github.io/notify-rust/.)"
+description = "A commandline tool that shows desktop notifications using [notify-rust](http://hoodie.github.io/notify-rust/)."
```
Comment (hoodie) [inline Cargo.toml:6]: nice catch

File: src/main.rs [hunk 2]
```diff
@@ -1,107 +1,102 @@
...
         #[cfg(all(unix, not(target_os = "macos")))]
-        {
+        Commands::Server => {
             use notify_rust::server::NotificationServer;
             use std::thread;
             let server = NotificationServer::create();
-            thread::spawn(move || NotificationServer::start(&server, |notification| println!("{:#?}", notification)));
+            thread::spawn(move || {
+                NotificationServer::start(&server, |notification| println!("{:#?}", notification))
+            });
```
Comment (hoodie) [inline src/main.rs:99]: I'm afraid this will have to go though, there is no server feature anymore in notify-rust 4.x

**Review by hoodie — COMMENTED** [review]
Thanks so much!

File: src/main.rs (hunk 2 above)
Comment (hybras) [reply src/main.rs:99]: I was wondering. Couldn't tell if I missed it due to feature flags / conditional compile / moved elsewhere, or it was actually gone

**Pull request merged**